### ISSUE TITLE
Only rebuild widget tree if network result.data is different from cached data

### ISSUE THREAD

**Issue by lazylazyllama** [zino-app/graphql-flutter#616]
**Is your feature request related to a problem? Please describe.**
If you use the `cacheAndNetwork` policy in combination with the Query widget, graphql_flutter will build your ui based on your cache first and then do a full rebuild, after the network fetch is finished. This can look a bit odd when you are dealing with big amounts of data like pdfs. A PDF takes some time to render, so the experience is kind of bad. 
What I mean is, that flutter will render the pdf based on the cache first, which takes like a second or so.. a few moments later the refetch will be finished and the pdf that just showed up will disappear just to appear again shortly after ... so it feels kind of buggy from the end users perspective.

**Describe the solution you'd like**
I want flutter_graphql to only rebuild the widget tree, if the data fetched from the network is different to the data that is already available inside the cache.

**Comment by vasilich6107** [comment]
IMHO cache items should be compared against each other with use of `DeepCollectionEquality` as far as cache records are not immutable. This could be time consuming. You can use samples from examples `flutter_bloc` https://github.com/zino-app/graphql-flutter/tree/master/examples/flutter_bloc and apply your desired behaviour inside listener
https://github.com/zino-app/graphql-flutter/blob/master/examples/flutter_bloc/lib/extended_bloc/graphql/bloc.dart#L17

**Comment by lazylazyllama** [comment]
I think the `Query()` Widget should have an additional property like `bool rerender(dynamic oldData, dynamic newData);` so that this can be easily opted in by the developer.

So that you could do something like this:
```dart
return Query(
    options: QueryOptions(
        document: document,
        rerender: (oldValue, newValue) => oldValue.toJson() == newValue.toJson(),
    ),
);
```

**Comment by micimize** [comment]
Closed by #754 and configurable by `alwaysRebroadcast`
